packages/react-core/src/components/EmptyState/EmptyStateHeader.tsx

```diff
@@ -0,0 +1,46 @@
+import * as React from 'react';
+import { css } from '@patternfly/react-styles';
+import styles from '@patternfly/react-styles/css/components/EmptyState/empty-state';
+import { EmptyStateIcon } from './EmptyStateIcon';
+
+export interface EmptyStateHeaderProps extends React.HTMLProps<HTMLDivElement> {
+  /** Content rendered inside the EmptyStateHeader */
+  children?: React.ReactNode;
+  /** Additional classes added to the EmptyStateHeader */
+  className?: string;
+  /** Additional classes added to the title inside EmptyStateHeader */
+  titleClassName?: string;
+  /** Additional classes added to the Icon inside EmptyStateHeader */
+  iconClassName?: string;
+  /** Text of the title inside EmptyStateHeader */
+  titleText?: string;
+  /** Icon component to be rendered inside EmptyStateHeader. Can be a Spinner component. */
+  icon?: React.ComponentType<any>;
+}
+
+export const EmptyStateHeader: React.FunctionComponent<EmptyStateHeaderProps> = ({
+  children,
+  className = '',
+  titleClassName = '',
+  iconClassName = '',
+  titleText,
+  icon,
+  ...props
+}: EmptyStateHeaderProps) => {
+  const emptyStateTitle = titleText ? (
+    <div className={css('pf-c-empty-state__title')}>
+      <h1 className={css(styles.emptyStateTitleText, titleClassName)}>{titleText}</h1>
```

Comment:
I don't think we should hardcode an `h1` here. The code editor for example uses the Title component with an h4 heading, so there may be cases where the Empty State title text should have something other than an h1 to match the surrounding heading hierarchy.

We could just create another prop similar to the Title component's `headingLevel` prop.